config/custom_dns.go

```diff
@@ -14,14 +14,56 @@ type CustomDNS struct {
 	RewriterConfig      `yaml:",inline"`
 	CustomTTL           Duration         `yaml:"customTTL" default:"1h"`
 	Mapping             CustomDNSMapping `yaml:"mapping"`
+	ZoneFileMapping     ZoneFileDNS      `yaml:"zoneFileMapping" default:""`
 	FilterUnmappedTypes bool             `yaml:"filterUnmappedTypes" default:"true"`
 }
 
 type (
 	CustomDNSMapping map[string]CustomDNSEntries
 	CustomDNSEntries []dns.RR
+
+	ZoneFileDNS CustomDNSMapping
 )
 
+func (z *ZoneFileDNS) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var input string
+	if err := unmarshal(&input); err != nil {
+		return err
+	}
+
+	result := make(ZoneFileDNS, len(input))
+
+	if input != "" {
+		zoneParser := dns.NewZoneParser(strings.NewReader(input), "", "")
```

Comment:
Let's put an arbitrary file name so errors make sense:
```suggestion
		zoneParser := dns.NewZoneParser(strings.NewReader(input), "", "<YAML config>")
```

Also could you try what setting the origin arg to an empty string does to a relative name?  
Basically does`www 3600 A 1.2.3.4` end up being `www.`, does it error, or something else?  
I'm wondering if it's something we should document or error on our side if no `$ORIGIN` was specified by the user and wet get a relative name.

Reply:
This returns an error. I have written a test to confirm that this remains the case, but I wouldn't think that this is something that the project would document.

Reply:
I agree. Good idea for the test!